plutus-core/plutus-core/src/PlutusCore/Subst.hs

```diff
@@ -147,65 +147,46 @@ termSubstFreeNames = purely termSubstFreeNamesA
 -- Free variables
 
 -- | Get all the free term variables in a term.
-fvTerm :: Ord name => Term tyname name uni fun ann -> Set name
-fvTerm = cata f
-  where
-    f (VarF _ n)        = singleton n
-    f (TyAbsF _ _ _ t)  = t
-    f (LamAbsF _ n _ t) = delete n t
-    f (ApplyF _ t1 t2)  = t1 `union` t2
-    f (TyInstF _ t _)   = t
-    f (UnwrapF _ t)     = t
-    f (IWrapF _ _ _ t)  = t
-    f ConstantF{}       = Set.empty
-    f BuiltinF{}        = Set.empty
-    f ErrorF{}          = Set.empty
+fvTerm :: Ord name => Traversal' (Term tyname name uni fun ann) name
```

Comment:
Because it was so close to being a `Traversal` that I thought I might as well! Also, I left it out of this PR because it was already too much, but with a little more work you can actually use this to replace the naive substitution functions that we have, which would let us delete further code.

Reply:
I wrote about it: https://www.michaelpj.com/blog/2022/06/12/lenses-for-tree-traversals-redux.html